### PR TITLE
Fix for recent version of OCaml, js_of_ocaml, lwt, core_kernel

### PR DESCRIPTION
**Pull request by hhugo**
Make the code compile with recent versions of core_kernel (v0.13.1), lwt (5.3.0), js_of_ocaml (3.6), ocaml (4.10)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -224,9 +224,18 @@ COPIED_FLOWLIB=\
 COPIED_PRELUDE=\
 	$(foreach lib,$(wildcard prelude/*.js),_build/$(lib))
 
+# $ ocamlfind query sedlex lwt lwt_log lwt.unix lwt_ppx unix str bigarray ppx_let core_kernel base wtf8 dtoa  -predicates javascript -o-format -r
+# Manualy fix path for +base/base_internalhash_types/runtime.js and +ppx_expect/collector/runtime.js
```
Comment (hhugo) [inline Makefile:228]: https://github.com/ocaml/dune/pull/3445

**Comment by mroch** [comment]
thanks for this! we have a lot of different environments to get this to build in so it might take a bit of time to merge this, or I might have to land it in a couple pieces.

**Comment by hhugo** [comment]
Not pressing for me. Im doing this as I might use the Js parser in js_of_ocaml at some point in the future.

Note that I would expect the changes I made to be backward compatible with old version of ocaml, core_kernel.

**Comment by hhugo** [comment]
How much of that PR has been integrated already ? Should I leave it open ? Or close it ?

**Comment by mroch** [comment]
thanks again for this and sorry for the horrible delay. internally, we were stuck on ocaml 4.07, which meant we couldn't upgrade base or js_of_ocaml to versions that supported 4.10. that's why this is taking so long, and why it's happening in slightly different ways (like js_of_ocaml 3.5.2). I just bumped again to js_of_ocaml 3.7.1.

I've been using this PR as a reference, so I appreciate that you left it open and rebased it. I'll take care of rebasing again and merging what's left... hopefully without any more internal blockers.

**Pull request closed without merging**